### PR TITLE
Create User Profile APIs

### LINKED ISSUE #12: UserProfile APIs

### PR DESCRIPTION
**Pull request by PCalape**

### CODE REVIEW

File: src/modules/user/dto/user-profile.input.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export class UserProfileInput {
```
Comment (japhethbalane) [inline src/modules/user/dto/user-profile.input.ts:1]: Do we still need this @philip-calape @arnellebalane ?
There is already a `UserProfile` class that always mirrors the `UserProfileInput`.

Comment (PCalape) [reply src/modules/user/dto/user-profile.input.ts:1]: Hello @japhethbalane yes we need to introduce a separate class for inputs the same with outputs in order to introduce a layer of separation between our models passed on our server. For now the input, output, and model classes are exactly the same but if ever we add another functionality which requires different input/output from our model to be used by our server, this separation will be helpful. Thanks!

**Review by japhethbalane — APPROVED** [review]
LGTM :+1:

File: src/modules/user/adapter/controller/user-profile.controller.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { UserProfileInput } from '../../dto/user-profile.input';
+import { UserProfileOutput } from '../../dto/user-profile.output';
+
+@Controller('/user-profile')
+export class UserProfileController {
+    constructor(private readonly userProfileService: UserProfileService) {}
+
+    @Post()
+    @UseGuards(SupabaseGuard)
+    async upsertUserProfile(@AuthUser() { sub: id }: Record<string,string>, @Body() userProfile: UserProfileInput): Promise<void> {
```
Comment (arnellebalane) [inline src/modules/user/adapter/controller/user-profile.controller.ts]: @PCalape The `Record<string,string>` types in this file, why not define it separate like `UserProfileInput`?

File: supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+alter table if exists user_profiles add constraint unique_user_id unique (user_id);
```
Comment (arnellebalane) [inline supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: @PCalape Let's just drop the `id` column and use `user_id` as the primary key.

**Review by arnellebalane — CHANGES_REQUESTED** [review]

File: src/modules/user/adapter/controller/user-profile.controller.ts (hunk 2 above)
Comment (PCalape) [reply src/modules/user/adapter/controller/user-profile.controller.ts]: Copy on this @arnellebalane!

File: supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql (hunk 3 above)
Comment (PCalape) [reply supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: Copy on this @arnellebalane!

File: src/modules/user/adapter/controller/user-profile.controller.ts (hunk 2 above)
Comment (PCalape) [reply src/modules/user/adapter/controller/user-profile.controller.ts]: Done applying changes @arnellebalane

File: supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql (hunk 3 above)
Comment (PCalape) [reply supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: Done applying changes @arnellebalane

Comment (arnellebalane) [reply supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: @PCalape You can now delete this migration file, this field is now a primary key in the next migration so no need to manually make it unique.

**Review by arnellebalane — CHANGES_REQUESTED** [review]

File: supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql (hunk 3 above)
Comment (PCalape) [reply supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: Nice catch @arnellebalane will apply the change

Comment (PCalape) [reply supabase/migrations/20231014104130_make_user_id_in_user_profiles_table_unique.sql]: Done applying the changes @arnellebalane

**Review by arnellebalane — APPROVED** [review]
LGTM thanks @PCalape 💯

**Pull request merged**